variables.tf

```diff
@@ -364,9 +364,30 @@ variable "addons" {
   nullable    = false
   default     = {}
 
+  ########################################################################################################################
+  # OCP AI Addon version validation
+  ########################################################################################################################
+
   validation {
-    condition     = (lookup(var.addons, "openshift-ai", null) != null ? lookup(var.addons["openshift-ai"], "version", null) == null : true) || (tonumber(local.ocp_version_num) >= 4.16)
-    error_message = "OCP AI add-on requires OCP version >= 4.16.0"
+    condition = (
+      lookup(var.addons, "openshift-ai", null) == null ||
+      lookup(var.addons["openshift-ai"], "version", null) == null ||
+      (
+        contains(keys(local.ocp_ai_addon_supported_versions), var.addons["openshift-ai"].version) &&
+        (
+        local.ocp_version_num >= tonumber(regexall("\\d+\\.\\d+", split(" ", local.ocp_ai_addon_supported_versions[var.addons["openshift-ai"].version].supported_openshift_range)[0])[0])) &&
+        (
+          local.ocp_version_num < tonumber(regexall("\\d+\\.\\d+", split(" ", local.ocp_ai_addon_supported_versions[var.addons["openshift-ai"].version].supported_openshift_range)[1])[0])
+        )
+      )
+    )
+
+    error_message = (
+      var.addons["openshift-ai"] != null && var.addons["openshift-ai"].version != null ?
```

Comment:
The validation should be handled properly in the error message too where the add-on is used as index else similar error message will be shown - 

<img width="800" height="197" alt="Image" src="https://github.com/user-attachments/assets/2f091aa7-434f-43a0-8699-ad922db5bb19" />